### PR TITLE
Audio Params

### PR DESCRIPTION
**Pull request by jens1o**
| Q                  | A
| ------------------ | ---
| Bug fix?           | no
| New feature?       | yes
| BC breaks?         | yes
| Deprecations?      | no
| License            | MIT

#### What's in this PR?

The option to define params, but also for the audio.
#### Why?
It's shitty to have one feature in a class but not in a class that's extending the other.
#### Example Usage

```php
$foo = new Mp3();

$foo->setAdditionalParameters(array('-foo', 'bar'));

```
#### BC Breaks/Deprecations
Needed to move the getter from the videointerface to the audiointerface.

### CODE REVIEW

**Comment by jens1o** [comment]
oh... I'd created the branch depending on another branch... That wasn't me!!!

File: src/FFMpeg/Media/Video.php [hunk 1]
```diff
@@ -121,11 +123,9 @@ public function save(FormatInterface $format, $outputPathfile)
...
-        if ($format instanceof VideoInterface) {
-            if (null !== $format->getAdditionalParameters()) {
-                foreach ($format->getAdditionalParameters() as $additionalParameter) {
-                    $commands[] = $additionalParameter;
-                }
+        if (null !== $format->getAdditionalParameters()) {
+            foreach ($format->getAdditionalParameters() as $additionalParameter) {
+                $commands[] = $additionalParameter;
             }
         }
```
Comment (Romain) [inline src/FFMpeg/Media/Video.php]: Shouldn't we put this:

```php
if (null !== $format->getAdditionalParameters()) {
             foreach ($format->getAdditionalParameters() as $additionalParameter) {
                 $commands[] = $additionalParameter;
              }
          }
```
inside the test `if ($format instanceof AudioInterface)` ?

**Comment by Romain** [comment]
Otherwise, I'm globally fine with these changes.

**Comment by Romain** [comment]
Oh, and can you just say in the description of your PR, which issue you're solving?

**Comment by Romain** [comment]
Oh, I just realized, you used a version of the bundle which is not merged yet.
You have the code of the #335 PR...

File: src/FFMpeg/Media/Video.php (hunk 1 above)
Comment (jens1o) [inline src/FFMpeg/Media/Video.php]: I didn't did that because it inherits it. The video interface extends the audio interface.

**Comment by jamesgraham** [comment]
@Romain does every format now need this `$audioParameters` attribute?

We are getting an error in the Wav one:

```Undefined property: FFMpeg\Format\Audio\Wav::$additionalParameters
```

**Comment by jens1o** [comment]
You can't access it directly @jamesgraham, it is protected. You need to use the setter.

**Comment by jamesgraham** [comment]
Hi @jens1o 

We aren't trying to access this attribute or set any parameters.

Our code looks a bit like this:

```
$audio->filters()->clip(TimeCode::fromSeconds($duration));
$audio->save(new Wav(), $trimmedTmpFile);
```

And the error now throws
```	
Undefined property: FFMpeg\Format\Audio\Wav::$additionalParameters
line	54
code	8
file	/home/vendor/php-ffmpeg/php-ffmpeg/src/FFMpeg/Format/Audio/DefaultAudio.php
```

**Comment by jamesgraham** [comment]
Maybe I am missing something but I don't see that attribute listed in the [DefaultAudio](https://github.com/PHP-FFMpeg/PHP-FFMpeg/blob/master/src/FFMpeg/Format/Audio/DefaultAudio.php) class.

Again we may be doing something wrong but I just wanted to ask before I dig deeper.

**Comment by jens1o** [comment]
What did i do... My bad...

Afaik php, calling `$audio->save((new Wav)->setAdditionalParameters(array()), $trimmedTmpFile);` should fix the error... Omg, how could this happen...

**Comment by jens1o** [comment]
Does it work for you? I'm creating an pr immediately, hopefully @Romain has some time to merge asap.

**Comment by jens1o** [comment]
I wonder why the tests passed successfully...

**Comment by jamesgraham** [comment]
Thanks @jens1o we have just pinned the version for now

**Comment by xparthx** [comment]
I also had the same error:

`Undefined property: FFMpeg\Format\Audio\Mp3::$additionalParameters`

Adding below in _php-ffmpeg/php-ffmpeg/src/FFMpeg/Format/Audio/DefaultAudio.php_ fixed the issue for me.

`protected $additionalParameters = null;`

**Comment by Romain** [comment]
I'm also having an error.
I'll check if I can rollback to 0.9.3.

**Comment by Romain** [comment]
@jamesgraham I've reverted this PR for the moment and published a new release without it.
I think we need to take time to check more in details how this PR was built and what was wrong in the code and in the tests. Tests properly written shouldn't have pass...
My bad on the review, I didn't detect this bug.

**Comment by Romain** [comment]
I tried the 0.9.5, it seems to work properly. If it's not the case, just let me know.

**Pull request merged**